spring-cloud-gcp-pubsub-stream-binder/src/test/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisionerTests.java

```diff
@@ -114,6 +114,29 @@ public void testProvisionConsumerDestination_specifiedGroupTopicInDifferentProje
 		verify(this.pubSubAdminMock).createSubscription("topic_A.group_A", "projects/differentProject/topics/topic_A");
 	}
 
+	@Test
+	public void testProvisionConsumerDestination_customSubscription() {
+		when(this.properties.getExtension()).thenReturn(this.pubSubConsumerProperties);
+		when(this.pubSubConsumerProperties.getCustomSubscription()).thenReturn("my-custom-subscription");
+
+		PubSubConsumerDestination result = (PubSubConsumerDestination) this.pubSubChannelProvisioner
+				.provisionConsumerDestination("topic_A", "group_A", this.properties);
+
+		assertThat(result.getName()).isEqualTo("my-custom-subscription");
+	}
+
+	@Test
+	public void testProvisionConsumerDestination_customSubscriptionDifferentProject() {
+		when(this.properties.getExtension()).thenReturn(this.pubSubConsumerProperties);
+		when(this.pubSubConsumerProperties.getCustomSubscription())
+				.thenReturn("projects/differentProject/subscriptions/my-custom-subscription");
+
+		PubSubConsumerDestination result = (PubSubConsumerDestination) this.pubSubChannelProvisioner
+				.provisionConsumerDestination("topic_A", "group_A", this.properties);
+
+		assertThat(result.getName()).isEqualTo("my-custom-subscription");
```

Comment:
So, this scenario is actually interesting. If we discard the different project prefix, will the binder will try to connect to `my-custom-subscription` in the wrong (current) project?

Reply:
Hmm... I think you're right - I assumed that the subscription created inside this method would used to create the subscriber, but I didn't look close enough.